### ISSUE TITLE
PulseAudio: Exiting/not connecting on an `Err` value: "SendError(..)

### ISSUE THREAD

**Issue by varakh** [Spotifyd/spotifyd#73]
Hi,

I have two errors which occur while using spotifyd on my Raspberry Pi v3 and I don't see a reason neither do I get obvious output about it. Besides having "Authentication failed" very often those two are making me crazy:

1. When I play music via Spotify App on my Android device and switch to my Raspberry Pi output Spotify App states "Connecting ..." and then aborts after 3 seconds. No output. Nothing.
2. When I don't play music and try to switch to Raspberry Pi output first and then press play on my Spotify App spotifyd process exits without any output. Process is just killed.

I tried to get more output with --verbose, but nothing special shows up, no ERROR tagged log line.

Any ideas?

**Comment by flokli** [comment]
'Authentication failed' errors seem to be caused due to the `system_information_string` being sent inside `librespot`. This seems to be fixed in upstream librespot. There is a PR waiting in #75 which should fix this in `spotifyd` too. I run a version with this patch for some days now without these problems.

Regarding your problems in switching from and to the Android Mobile app, I wasn't able to reproduce with the desktop app. I can check again with the mobile app in some days.

**Comment by varakh** [comment]
I have some news about this. Updated with latest librespot etc and Auth errors are gone.

But the main problem remains: I compiled it on my homeserver with pulseaudio backend support. If I start playing a song with Spotify Connect on the homeserver being the active "player" I get the following and spotifyd crashes:

`13:31:33 [ERROR] Caught panic with message: called `Result::unwrap()` on an `Err` value: "SendError(..)"`

With the same compiled binary copied from my homeserver, desktop is working just fine. I think that this might be pulseaudio backend related?

Unfortunately I need pulseaudio support in order to bind spotifyd output to a pulseaudio sink.
